applications/StructuralMechanicsApplication/custom_elements/timoshenko_beam_element_2D2N.cpp

```diff
@@ -0,0 +1,1051 @@
+        noalias(rRHS) -= global_size_N * V * jacobian_weight;
+
+        // Now we add the body forces contributions
+        global_size_N.clear();
+        global_size_N[0] = N_u[0];
+        global_size_N[3] = N_u[1];
+        noalias(rRHS) += global_size_N * local_body_forces[0] * jacobian_weight * GetProperties()[CROSS_AREA];
+        global_size_N.clear();
+        global_size_N[1] = N_shape[0];
+        global_size_N[2] = N_shape[1];
+        global_size_N[4] = N_shape[2];
+        global_size_N[5] = N_shape[3];
+        noalias(rRHS) += global_size_N * local_body_forces[1] * jacobian_weight * GetProperties()[CROSS_AREA];
+    }
+
+    RotateRHS(rRHS, r_geometry);
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void TimoshenkoBeamElement2D2N::RotateLHS(
+    MatrixType& rLHS,
+    const GeometryType& rGeometry
+)
+{
+    const double angle = StructuralMechanicsElementUtilities::
+        GetReferenceRotationAngle2D2NBeam(GetGeometry());
+    if (std::abs(angle) > std::numeric_limits<double>::epsilon()) {
+        BoundedMatrix<double, 3, 3> T, Tt;
+        StructuralMechanicsElementUtilities::BuildRotationMatrixFor2D2NBeam(T, angle);
+        noalias(Tt) = trans(T);
+
+        // We rotate each submatrix independently
+        RangeMatrixType rLHSaa(rLHS, boost::numeric::ublas::range(0, 3), boost::numeric::ublas::range(0, 3));
+        RangeMatrixType rLHSab(rLHS, boost::numeric::ublas::range(0, 3), boost::numeric::ublas::range(3, 6));
+        RangeMatrixType rLHSba(rLHS, boost::numeric::ublas::range(3, 6), boost::numeric::ublas::range(0, 3));
+        RangeMatrixType rLHSbb(rLHS, boost::numeric::ublas::range(3, 6), boost::numeric::ublas::range(3, 6));
+
+        rLHSaa = prod(T, MatrixType(prod(rLHSaa, Tt)));
+        rLHSab = prod(T, MatrixType(prod(rLHSab, Tt)));
+        rLHSba = prod(T, MatrixType(prod(rLHSba, Tt)));
+        rLHSbb = prod(T, MatrixType(prod(rLHSbb, Tt)));
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void TimoshenkoBeamElement2D2N::RotateRHS(
+    VectorType& rRHS,
+    const GeometryType& rGeometry
+)
+{
+    const double angle = StructuralMechanicsElementUtilities::
+        GetReferenceRotationAngle2D2NBeam(GetGeometry());
+    if (std::abs(angle) > std::numeric_limits<double>::epsilon()) {
+        BoundedMatrix<double, 3, 3> T;
+        StructuralMechanicsElementUtilities::BuildRotationMatrixFor2D2NBeam(T, angle);
+
+        VectorType rRHSa(3);
+        rRHSa[0] = rRHS[0];
+        rRHSa[1] = rRHS[1];
+        rRHSa[2] = rRHS[2];
+        VectorType rRHSb(3);
+        rRHSb[0] = rRHS[3];
+        rRHSb[1] = rRHS[4];
+        rRHSb[2] = rRHS[5];
+
+        rRHSa = prod(T, rRHSa);
```

Comment:
i would avoid use the aliased case, as it will create a temporary.

you could write 

~~~cpp
VectorType rRHSa=prod(T,rRHS(range...))
~~~